woocommerce/wc-admin
# Update component Table to align texts and numeric a bit smarter

Comes from this issue in WCPay https://github.com/Automattic/woocommerce-payments/issues/2351

### Description 

This addresses this concern from @LevinMedia on p1623896858155800/1623882603.153200-slack-CGGCLBN58

> This has been a nagging issue in wc admin analytics tables for some time. We should right align numbers, and text align left.

What changes specifically: 

- If `isLeftAligned` is set, just use it. 
- If `isLeftAligned` is not set, its value will fall back to this value `! isNumeric`, i.e. text fields are left-aligned and numeric fields are right-aligned.

Benefits: 

- Compatibility with the current table fields have already set `isLeftAligned`. E.g. [`Date` field](https://github.com/woocommerce/woocommerce-admin/blob/f91140673aba20075f2b26aacdbb2be94aef8834/client/analytics/report/orders/table.js#L34-L41) in Analytics > Orders 
- Reduce the need of declaring `isLeftAligned` but still address @LevinMedia's concern nicely. 

### Accessibility




### Screenshots

| Before  | After |
| ------------- | ------------- |
| ![before](https://user-images.githubusercontent.com/10045087/127292874-baa8fc40-1a11-48c0-9acb-40bc4e9e93ee.png) | ![after](https://user-images.githubusercontent.com/10045087/127292932-3b9c9ed7-bcd4-46ab-b518-c3dcc606f512.png) |

### Detailed test instructions:

1. See the current state of tables in Analytics.
2. Apply this PR. 
3. Make sure to rebuild the plugin. `npm start`. 
4. See the change and expect that table fields will be aligned correctly. 

## Extra checks : 

If the concept of this PR is approved, do two following tasks: 

- [ ] Review all tables under Analytics. 
- [ ] Align fields that do not look right by using `isLeftAligned`.
